packages/sdk/src/dataProtector/sharing/removeProtectedDataFromRenting.ts

```diff
@@ -1,71 +1,96 @@
-import { WorkflowError } from '../../utils/errors.js';
-import { throwIfMissing } from '../../utils/validators.js';
+import { GraphQLClient } from 'graphql-request';
+import { DEFAULT_SHARING_CONTRACT_ADDRESS } from '../../config/config.js';
+import { ErrorWithData, WorkflowError } from '../../utils/errors.js';
+import {
+  addressOrEnsOrAnySchema,
+  throwIfMissing,
+} from '../../utils/validators.js';
 import {
   IExecConsumer,
   RemoveProtectedDataFromRentingParams,
   SuccessWithTransactionHash,
   SubgraphConsumer,
+  Address,
 } from '../types/index.js';
 import { getSharingContract } from './smartContract/getSharingContract.js';
-import {
-  collectionExists,
-  isCollectionOwner,
-  isProtectedDataInCollection,
-} from './utils.js';
+import { getProtectedDataById } from './subgraph/getProtectedDataById.js';
 
 export const removeProtectedDataFromRenting = async ({
   iexec = throwIfMissing(),
   graphQLClient = throwIfMissing(),
-  collectionTokenId = throwIfMissing(),
   protectedDataAddress = throwIfMissing(),
 }: IExecConsumer &
   SubgraphConsumer &
   RemoveProtectedDataFromRentingParams): Promise<SuccessWithTransactionHash> => {
-  //TODO:Input validation
-  const collectionExist = await collectionExists({
-    graphQLClient,
-    collectionTokenId: collectionTokenId,
-  });
-  if (!collectionExist) {
-    throw new WorkflowError(
-      'Failed to Remove Protected Data From Renting: collection does not exist.'
-    );
-  }
+  const vProtectedDataAddress = addressOrEnsOrAnySchema()
```

Comment:
Can a protected data really be an ENS here? I'm really not sure.

Reply:
yeah endUser can set ENS on their assets as we made for the web3mail Dapp. I do that to be uniforme with other V1 function (processProtectedData, ...)
May be we should refactor this validator name & code

Reply:
👍